data/lua/core/gui.lua

```diff
@@ -154,3 +154,6 @@ wesnoth.get_dialog_value = wesnoth.deprecate_api('wesnoth.get_dialog_value', val
 wesnoth.add_dialog_tree_node = wesnoth.deprecate_api('wesnoth.add_dialog_tree_node', '<widget>:add_item_of_type', 1, nil, add_dialog_tree_node)
 wesnoth.remove_dialog_item = wesnoth.deprecate_api('wesnoth.remove_dialog_item', '<widget>:remove_items_at', 1, nil, remove_dialog_item)
 wesnoth.show_dialog = wesnoth.deprecate_api('wesnoth.show_dialog', 'gui.show_dialog', 1, nil, gui.show_dialog)
+if wesnoth.kernel_type() == "Game Lua Kernel" then
```

Comment:
Why restrict specifically this to Game Lua Kernel, what's different compared to `show_dialog`?

Reply:
That's a very good question. It has always been restricted to the Game Lua Kernel, so I simply didn't change that, but I'm not sure that there's a good reason for that…